pkg/kubernetes/scripts/virtual-machines/components/VmActions.jsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// @flow
+
+import React, { PropTypes } from 'react';
+import { gettext as _ } from 'cockpit';
+import { connect } from 'react-redux'
+
+import type { Vm } from '../types.jsx';
+import { vmIdPrefx } from '../utils.jsx';
+import { vmShutdown } from '../kube-middleware.jsx';
+import { vmActionFailed } from '../action-creators.jsx';
+
+// import DropdownButtons from '../../../../machines/components/dropdownButtons.jsx';
+import { mouseClick } from '../utils.jsx';
+
+React;
+
+function canShutdown({ vm }: { vm: Vm }) {
+  vm ; // TODO: check on state
+  return true;
```

Comment:
When would kubevirt not be able to shutdown a machine? Is that just to prevent race conditions when the button gets clicked twice? If that can happen, this should be implemented before merging.

Reply:
oops, good catch.
The idea is to provide a predicate when the `Shut Down` (or recently `Delete`) button is available.